12-version_control.Rmd

```diff
@@ -0,0 +1,821 @@
+(Figure \@ref(fig:create-new-file-02)). 
+Note that GitHub understands and renders `.md` files using a 
+[markdown syntax](https://guides.github.com/pdfs/markdown-cheatsheet-online.pdf) 
+very similar to Jupyter notebooks, so the "Preview" tab is especially helpful 
+for checking markdown code correctness.
+
+```{r create-new-file-02, fig.cap = 'New plain text files require a filename be inputted in the textbox circled in red, and file content entered in the larger text box (red arrow).', fig.retina = 2}
+image_read("img/version_control/create-new-file_02.png") %>%
+  image_flop() %>%
+  image_crop("3584x1300") %>%
+  image_flop()
+```
+
+Save and commit your changes by click the green "Commit changes" button at the 
+bottom of the page (Figure \@ref(fig:create-new-file-03)).
+
+```{r create-new-file-03, fig.cap = 'To be saved, newly created files are required to be committed along with an associated commit message.', fig.retina = 2}
+image_read("img/version_control/create-new-file_03.png") %>%
+  image_crop("3584x1500+1+500")
+```
+
+You can also upload files that you have created on your local machine by using 
+the "Add file" drop down menu and selecting "Upload files"
+(Figure \@ref(fig:upload-files-01)).
+
+```{r upload-files-01, fig.cap = 'New files of any type can be uploaded to GitHub.', fig.retina = 2}
+image_read("img/version_control/upload-files_01.png") %>%
+  image_flop() %>%
+  image_crop("3584x1600") %>%
+  image_flop()
+```
+
+To select the files from your local computer to upload, you can either drag and 
+drop them into the grey box area shown below, or click the "choose your files" 
+link to access a file browser dialog. Once the files you want to upload have 
+been selected, click the green "Commit changes" button at the bottom of the 
+page (Figure \@ref(fig:upload-files-02)).
+
+```{r upload-files-02, fig.cap = 'Uploaded files can specified by dragging them into the GitHub website where specified (red circle) or by clicking on the "choose your files" to select them via a file browser. Uploaded files are also required to be committed along with an associated commit message.', fig.retina = 2}
+image_read("img/version_control/upload-files_02.png") %>%
+  image_flop() %>%
+  image_crop("3584x2200") %>%
+  image_flop()
+```
+
+Note that Git and GitHub are designed to track changes in individual files. 
+*Do not* upload your whole project in an archive file (e.g. `.zip`), because
+then Git can only keep track of changes to the entire `.zip` file---that 
+wouldn't be very useful if you're trying to see the history of changes to a 
+single code file in your project!
+
+
+## Cloning your repository on JupyterHub
+
+Although there are several ways to create and edit files on 
+GitHub, they are not quite powerful enough for 
+efficiently creating and editing complex files, or files that need to be 
+executed to assess whether they work (e.g., files containing code). Thus, it is 
```

Comment:
Could we write this first sentence in more plain language or give an example? e.g. "You wouldn't be able to run an R code directly on GitHub"

Reply:
done in 129a8de